From 8ea8cdc1c626d3b9fcc763958ab66833d1ec0274 Mon Sep 17 00:00:00 2001
From: Nicolas Toromanoff <nicolas.toromanoff@foss.st.com>
Date: Thu, 31 Mar 2022 09:31:23 +0200
Subject: [PATCH 391/529] ARM: dts: stm32: add dmas entries for CRYP driver

The CRYP IP can use the MDMA to transfer in and out buffers.
The DMA can be used in CRYP for stm32mp13xc, stm32mp13xf, stm32mp15xc,
stm32mp15xf and can be enabled by default to gain performance.

Signed-off-by: Nicolas Toromanoff <nicolas.toromanoff@foss.st.com>
Change-Id: Ic552beb9046ac3ff855a65bc960be5f301bd4c69
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/246343
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas TOROMANOFF <nicolas.toromanoff@st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Lionel DEBIEVE <lionel.debieve@foss.st.com>
Tested-by: Nicolas TOROMANOFF <nicolas.toromanoff@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/280579
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Thomas BOURGOIN <thomas.bourgoin@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 arch/arm/boot/dts/stm32mp13xc.dtsi | 3 +++
 arch/arm/boot/dts/stm32mp13xf.dtsi | 3 +++
 arch/arm/boot/dts/stm32mp15xc.dtsi | 3 +++
 arch/arm/boot/dts/stm32mp15xf.dtsi | 3 +++
 4 files changed, 12 insertions(+)

--- a/arch/arm/boot/dts/stm32mp13xc.dtsi
+++ b/arch/arm/boot/dts/stm32mp13xc.dtsi
@@ -12,6 +12,9 @@
 			interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc CRYP1>;
 			resets = <&rcc CRYP1_R>;
+			dmas = <&mdma 28 0x0 0x400202 0x0 0x0 0x0>,
+			       <&mdma 29 0x3 0x400808 0x0 0x0 0x0>;
+			dma-names = "in", "out";
 			status = "disabled";
 		};
 	};
--- a/arch/arm/boot/dts/stm32mp13xf.dtsi
+++ b/arch/arm/boot/dts/stm32mp13xf.dtsi
@@ -12,6 +12,9 @@
 			interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc CRYP1>;
 			resets = <&rcc CRYP1_R>;
+			dmas = <&mdma 28 0x0 0x400202 0x0 0x0 0x0>,
+			       <&mdma 29 0x3 0x400808 0x0 0x0 0x0>;
+			dma-names = "in", "out";
 			status = "disabled";
 		};
 	};
--- a/arch/arm/boot/dts/stm32mp15xc.dtsi
+++ b/arch/arm/boot/dts/stm32mp15xc.dtsi
@@ -14,6 +14,9 @@
 			interrupts = <GIC_SPI 79 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc CRYP1>;
 			resets = <&rcc CRYP1_R>;
+			dmas = <&mdma1 29 0x0 0x400202 0x0 0x0 0x0>,
+			       <&mdma1 30 0x3 0x400808 0x0 0x0 0x0>;
+			dma-names = "in", "out";
 			status = "disabled";
 		};
 	};
--- a/arch/arm/boot/dts/stm32mp15xf.dtsi
+++ b/arch/arm/boot/dts/stm32mp15xf.dtsi
@@ -14,6 +14,9 @@
 			interrupts = <GIC_SPI 79 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc CRYP1>;
 			resets = <&rcc CRYP1_R>;
+			dmas = <&mdma1 29 0x0 0x400202 0x0 0x0 0x0>,
+			       <&mdma1 30 0x3 0x400808 0x0 0x0 0x0>;
+			dma-names = "in", "out";
 			status = "disabled";
 		};
 	};
